src/genobs/include/genobs/properties.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2010-2020 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef OBSERVABLES_PROPERTIES_HPP
+#define OBSERVABLES_PROPERTIES_HPP
+
+namespace GenObs {
+template <class DoF> struct traits;
+
+namespace detail {
+template <class T> struct decay { using type = typename std::decay_t<T>; };
```

Comment:
What's this for?

Reply:
type deduction for the traits... so you don't have to match pointer or ref to ```type``` because it will decay to ```type```

Reply:
the `typename` is not needed I think